internal/pkg/remote/endpoint/service.go

```diff
@@ -109,6 +121,16 @@ func (s *service) Status() (version string, err error) {
 	return vRes.Version, nil
 }
 
+// configVal returns the value of the specified key (if present), in the
+// service's additional known configuration.
+func (s *service) configVal(key string) string {
+	val, ok := s.configMap[key]
```

Comment:
Not sure this pattern is necessary here - doesn't a `map[string]string` already return an empty string upon lookup with a nonexistent key...?